File: DataBinding2WayJava/app/src/main/java/com/websarva/wings/android/databinding2wayjava/ui/MainViewModel.java
package com.websarva.wings.android.databinding2wayjava.ui;

import androidx.lifecycle.ViewModel;

import com.websarva.wings.android.databinding2wayjava.data.model.Rectangle;

public class MainViewModel extends ViewModel {
	private Rectangle _rectangle = new Rectangle();

	public MainViewModel() {
		generateNewHeight();
		generateNewWidth();
	}

	public void generateNewHeight() {
		int height = (int) (Math.random() * 10) + 1;
		_rectangle.setHeight(height);
	}

	public void generateNewWidth() {
		int width = (int) (Math.random() * 10) + 1;
		_rectangle.setWidth(width);
	}

	public Rectangle getRectangle() {
		return _rectangle;
	}
}

File: DataBinding2WayKotlin/app/src/main/java/com/websarva/wings/android/databinding2waykotlin/ui/MainViewModel.kt
package com.websarva.wings.android.databinding2waykotlin.ui

import androidx.lifecycle.ViewModel
import com.websarva.wings.android.databinding2waykotlin.data.model.Rectangle

class MainViewModel : ViewModel() {
	var rectangle: Rectangle = Rectangle()

	init {
		generateNewHeight()
		generateNewWidth()
	}

	fun generateNewHeight() {
		rectangle.height = (Math.random() * 10).toInt() + 1
	}

	fun generateNewWidth() {
		rectangle.width = (Math.random() * 10).toInt() + 1
	}
}
